src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -456,116 +484,117 @@ private async Task AcceptStreamsAsync()
             {
-                await using (stream.ConfigureAwait(false))
-                {
-                    if (stream.CanWrite)
-                    {
-                        // Server initiated bidirectional streams are either push streams or extensions, and we support neither.
-                        throw HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.StreamCreationError);
-                    }
-
-                    buffer = new ArrayBuffer(initialSize: 32, usePool: true);
-
-                    int bytesRead;
+                buffer = new ArrayBuffer(initialSize: 32, usePool: true);
+                int bytesRead;
 
-                    try
-                    {
-                        bytesRead = await stream.ReadAsync(buffer.AvailableMemory, CancellationToken.None).ConfigureAwait(false);
-                    }
-                    catch (QuicException ex) when (ex.QuicError == QuicError.StreamAborted)
-                    {
-                        // Treat identical to receiving 0. See below comment.
-                        bytesRead = 0;
-                    }
+                try
+                {
+                    // webtransport has a header of 3 bytes and in order to avoid reading
+                    // user content for webtransport stream we need at first to read 3 bytes
+                    bytesRead = await stream.ReadAsync(buffer.AvailableMemorySliced(3), CancellationToken.None).ConfigureAwait(false);
+                }
+                catch (QuicException ex) when (ex.QuicError == QuicError.StreamAborted)
+                {
+                    // Treat identical to receiving 0. See below comment.
+                    bytesRead = 0;
+                }
 
-                    if (bytesRead == 0)
-                    {
-                        // https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-unidirectional-streams
-                        // A sender can close or reset a unidirectional stream unless otherwise specified. A receiver MUST
-                        // tolerate unidirectional streams being closed or reset prior to the reception of the unidirectional
-                        // stream header.
-                        return;
-                    }
+                if (bytesRead == 0)
+                {
+                    // https://quicwg.org/base-drafts/draft-ietf-quic-http.html#name-unidirectional-streams
+                    // A sender can close or reset a unidirectional stream unless otherwise specified. A receiver MUST
+                    // tolerate unidirectional streams being closed or reset prior to the reception of the unidirectional
+                    // stream header.
+                    return;
+                }
 
+                buffer.Commit(bytesRead);
+                bool foundStreamType = VariableLengthIntegerHelper.TryRead(buffer.ActiveSpan, out long streamType, out bytesRead);
+                while (foundStreamType is false)
```

Comment:
I believe there is a potential problem with this loop. Can you think of it?

Reply:
The fact that I use bytesRead for both reading the stream and reading the number. Also if the encoding was done wrong from the server then we can stay in that loop forever. I think I will change the variables names and I will keep bytesRead for the bytes we receive from the stream and I will add a new variable named `bytesDecoded` for the decoding of the number and the while would be until we read <8 bytes from the stream or if we did find the stream type

Reply:
You should also stop if the server would suddenly gracefully close the stream before you reach 8 bytes (ReadAsync will start returning 0) -- if you don't check it, even "while <8 bytes" loop might become endless